bundle/CMakeLists.txt

```diff
@@ -0,0 +1,92 @@
+if(NOT CMAKE_BUILD_TYPE)
+  set(CMAKE_BUILD_TYPE "RelWithDebInfo" CACHE STRING 
+      "Choose the type of build, options are: Debug Release RelWithDebInfo." FORCE)
+endif()
+
+if(NOT BUILD_SHARED_LIBS)
+  set(BUILD_SHARED_LIBS ON CACHE BOOL "Build shared libraries." FORCE)
+endif()
+
+if(NOT CMAKE_INSTALL_PREFIX)
+  set(CMAKE_INSTALL_PREFIX "${CMAKE_SOURCE_DIR}/install" CACHE STRING
+      "Install path prefix, prepended onto install directories." FORCE)
+endif()
+
+project(serialbox2-all C CXX Fortran)
+enable_testing()
+cmake_minimum_required(VERSION 3.8.1)
+
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake")
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/../cmake")
+
+include(serialbox2CloneRepository)
+
+set(YODA_GIT_URL "https:///github.com/cosunae/yoda.git" 
+    CACHE PATH "URL of the yoda git repository to clone")
+set(YODA_GIT_BRANCH "create_library" CACHE STRING "Branch of the yoda git repository to clone")
+
+serialbox2_clone_repository(NAME yoda URL ${YODA_GIT_URL} BRANCH ${YODA_GIT_BRANCH} SOURCE_DIR YODA_SOURCE_DIR )
+
+list(APPEND CMAKE_MODULE_PATH "${YODA_SOURCE_DIR}/cmake")
+include(yodaInit)
+
+yoda_init()
+
+# Add custom targets
+yoda_add_target_clean_all(
+  "${CMAKE_BINARY_DIR}/prefix"
+  "${CMAKE_BINARY_DIR}/thirdparty"  
+  "${CMAKE_BINARY_DIR}/yoda-cmake"
+  "${CMAKE_BINARY_DIR}/boost"
+  "${CMAKE_BINARY_DIR}/boost-prefix"
+)
+
+# Set the default download directory (define YODA_DOWNLOAD_DIR)
+yoda_set_download_dir()
+include("thirdparty/Serialbox2AddBoost")
+
+set(serialbox2_cmake_args -DBOOST_ROOT=${BOOST_ROOT})
+set(serialbox2_dependencies "boost")
+
+yoda_find_package(
+  PACKAGE serialbox2
+  NO_DEFAULT_PATH
+  FORWARD_VARS 
+    BINARY_DIR serialbox2_binary_dir
+  DEPENDS ${serialbox2_dependencies}
+  ADDITIONAL
+    SOURCE_DIR "${CMAKE_SOURCE_DIR}/../"
+    YODA_ROOT "${YODA_SOURCE_DIR}"
+    CMAKE_ARGS 
+        ${serialbox2_cmake_args}
+)
+
+ExternalProject_Add(serialbox-test

```

Comment:
Finally, at this point, all the dependencies and serialbox itself are processed, compiled and install. 
Here we add a final CMake package as a external project that will load the serialbox CMake config file installed and verify that all the targets have been exported.